token/token.go

```diff
@@ -0,0 +1,158 @@
+	Expiration  string `json:"expiration"`
+	S3Config    string `json:"s3config"`
+}
+
+func readRequestBody(body io.ReadCloser) (tokenRequest tokenRequest, err error) {
+
+	reqBody, err := ioutil.ReadAll(body)
+	if err != nil {
+		return tokenRequest, fmt.Errorf("Reading request body")
+	}
+	defer body.Close()
+
+	err = json.Unmarshal(reqBody, &tokenRequest)
+	if err != nil {
+		return tokenRequest, fmt.Errorf("Unmarshaling data")
+	}
+
+	if tokenRequest.ProjectID == "" || tokenRequest.Swamid == "" {
+		return tokenRequest, fmt.Errorf("Incomplete data")
+	}
+
+	return tokenRequest, nil
+}
+
+func createECToken(key *ecdsa.PrivateKey, username string) (string, error) {
+	// signing method of token
+	token := jwt.New(jwt.SigningMethodES256)
+	// token headers
+	token.Header["alg"] = "ES256"
+	// token claims
+	claims := make(jwt.MapClaims)
+	claims["iss"] = helpers.Config.Iss
+
+	claims["exp"] = time.Now().AddDate(0, 0, helpers.Config.ExpirationDays).Unix()
+	claims["sub"] = username
+	token.Claims = claims
+
+	// create token
+	tokenString, err := token.SignedString(key)
+	if err != nil {
+		return "no-token", err
+	}
+
+	return tokenString, nil
+}
+
+func createS3Config(username string) (s3config string, expiration string, err error) {
+	s3config = "guess_mime_type = True \nhuman_readable_sizes = True\nuse_https = True\n" +
+		"multipart_chunk_size_mb = 50\n" + "check_ssl_certificate = False\n" +
+		"check_ssl_hostname = False\n" + "encoding = UTF-8\n" + "encrypt = False\n" +
+		"socket_timeout = 30\n"
+
+	prKeyParsed, err := helpers.ParsePrivateECKey(helpers.Config.PathToKey)
+	if err != nil {
+		log.Errorf("Probably the alg which is used as argument (ec) is different from the private's key alg (rsa):\n %v", err)
+	}
+
+	token, err := createECToken(prKeyParsed, username)
+	if err != nil {
+		log.Error(err)
+	}
+
+	expiration = time.Now().AddDate(0, 0, helpers.Config.ExpirationDays).Format("01-02-2006 15:04:05")
+	s3config += "secret_key = " + username + "\naccess_key = " + username + "\naccess_token = " + token +
+		"\nhost_base = " + helpers.Config.S3URL + "\nhost_bucket = " + helpers.Config.S3URL
+
+	s3config = b64.StdEncoding.EncodeToString([]byte(s3config))
+
+	return s3config, expiration, nil
+}
+
+func createResponse(tokenRequest tokenRequest, username string) (tokenResponse tokenResponse, err error) {
+
+	tokenResponse.RequestTime = time.Now().Format("01-02-2006 15:04:05")
+	tokenResponse.Swamid = tokenRequest.Swamid
+	tokenResponse.ProjectID = tokenRequest.ProjectID
+
+	tokenResponse.S3Config, tokenResponse.Expiration, err = createS3Config(username)
+	if err != nil {
+		return tokenResponse, fmt.Errorf("Error creating S3 configuration")
+	}
+
+	return tokenResponse, err
+}
+
+// getEGABoxAccount checks whether the access for the specified swamID and project is granted
+// and returns the respective ega-box account
+func getEGABoxAccount(projectID string, swamID string) (username string, err error) {
+
+	username = "ega-box-8902"
```

Comment:
This shouldn't be hard coded

Reply:
This function will in the future check against some database connecting the swamID with the ega-box accounts. Given that we don't have that, I thought I could return one real account. Also, this won't be part of the configuration in general, so I wanted to avoid adding it there. What do you think?

Reply:
Still it is bad to hardcode a username, specifically since that is one we use.
Since it needs to be mocked we can add it to the config for now.